rules/clients.go

```diff
@@ -0,0 +1,141 @@
+package rules
+
+import (
+	"bytes"
+	"net"
+	"sort"
+)
+
+type clients struct {
+	hosts []string
+	nets  ipNets
+}
+
+func (c *clients) Len() int {
+	if c == nil {
+		return 0
+	}
+	return len(c.hosts) + len(c.nets)
+}
+
+func (c *clients) Equal(other *clients) bool {
+	if c == nil {
+		return other == nil
+	}
+	if other == nil {
+		return false
+	}
+
+	if !stringArraysEquals(c.hosts, other.hosts) {
+		return false
+	}
+
+	if len(c.nets) != len(other.nets) {
+		return false
+	}
+
+	for i, subnet := range c.nets {
+		if subnet.String() != other.nets[i].String() {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (c *clients) finalize() {
+	if c != nil {
+		sort.Strings(c.hosts)
+		sort.Sort(c.nets)
+	}
+}
+
+// c != nil
+func (c *clients) add(client string) {
+	_, subnet, err := net.ParseCIDR(client)
+	if err == nil {
+		c.nets = append(c.nets, *subnet)
+		return
+	}
+
+	ip := net.ParseIP(client)
+	if ip != nil {
+		mask := net.CIDRMask(32, 32)
+		if ip.To4() == nil {
+			mask = net.CIDRMask(128, 128)
+		}
+		c.nets = append(c.nets, net.IPNet{IP: ip, Mask: mask})
+		return
+	}
+
+	c.hosts = append(c.hosts, client)
+}
+
+func newClients(clientStrs ...string) *clients {
+	c := &clients{}
+	for _, clientStr := range clientStrs {
+		c.add(clientStr)
+	}
+	c.finalize()
+	return c
+}
+
+func (c *clients) contains(host string, ipStr string) bool {
+	if c == nil {
+		return false
+	}
+
+	if findSorted(c.hosts, host) != -1 {
+		return true
+	}
+
+	ip := net.ParseIP(ipStr)
+	if ip != nil {
+		for _, net := range c.nets {
```

Comment:
`net` shadows package `net` (it's strange that golangci-lint does not highlight this)